pkg/virt-controller/watch/snapshot/restore.go

```diff
@@ -660,24 +660,25 @@ func patchVM(vm *kubevirtv1.VirtualMachine, patches []string) (*kubevirtv1.Virtu
 
 	marshalledVM, err := json.Marshal(vm)
 	if err != nil {
-		return nil, fmt.Errorf("cannot marshall VM %s: %v", vm.Name, err)
+		return vm, fmt.Errorf("cannot marshall VM %s: %v", vm.Name, err)
 	}
 
 	jsonPatch := "[\n" + strings.Join(patches, ",\n") + "\n]"
 
 	patch, err := jsonpatch.DecodePatch([]byte(jsonPatch))
 	if err != nil {
-		return nil, fmt.Errorf("cannot decode vm patches %s: %v", jsonPatch, err)
+		return vm, fmt.Errorf("cannot decode vm patches %s: %v", jsonPatch, err)
 	}
 
 	modifiedMarshalledVM, err := patch.Apply(marshalledVM)
 	if err != nil {
-		return nil, fmt.Errorf("failed to apply patch for VM %s: %v", jsonPatch, err)
+		return vm, fmt.Errorf("failed to apply patch for VM %s: %v", jsonPatch, err)
 	}
 
+	vm = &kubevirtv1.VirtualMachine{}
 	err = json.Unmarshal(modifiedMarshalledVM, vm)
 	if err != nil {
-		return nil, fmt.Errorf("cannot unmarshal modified marshalled vm %s: %v", string(modifiedMarshalledVM), err)
+		return vm, fmt.Errorf("cannot unmarshal modified marshalled vm %s: %v", string(modifiedMarshalledVM), err)
```

Comment:
The reason is that [this error message](https://github.com/kubevirt/kubevirt/blob/main/pkg/virt-controller/watch/snapshot/restore.go#L548) uses the VM's name as part of the error. If we return nil here it would lead to nil dereference.